urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/yuccasrv/socketmngr/ServerSocketThread.java

```diff
@@ -0,0 +1,468 @@
+		if( count == null )
+		    count = new BasicNumberType( 1 );
+		else
+		    count = new BasicNumberType( count.getInt() + 1 );
+		this.bindSettings.put( Constants.KEY_CONNECTION_COUNT, count );
+		    
+		//System.out.println( getClass().getName() + ".run() Telling observer about incoming TCP connection ..." );
+		this.observer.incomingTCPConnection( this, sock );
+		//System.out.println( getClass().getName() + ".run() Done." );
+		
+		
+	    }
+	    // END while: server was closed or thread was interrupted
+
+	    if( !this.serverSocket.isClosed() ) 
+		closeServerSocket();
+	    else //if( !this.isInterrupted() )
+		this.observer.serverSocketClosed( this );
+
+	} catch( SocketTimeoutException e ) {
+
+	    if( !this.isInterrupted() )
+		this.observer.serverSocketException( this, e );
+
+	} catch( IOException e ) {
+	    
+	    if( !this.isInterrupted() )
+		this.observer.serverSocketException( this, e );
+
+	} catch( SecurityException e ) {
+	    
+	    if( !this.isInterrupted() )
+		this.observer.serverSocketException( this, e );
+
+	} catch( IllegalBlockingModeException e ) {
+
+	    if( !this.isInterrupted() )
+		this.observer.serverSocketException( this, e );
+	    
+	}
+	
+    }
+
+    public void interrupt() {
+
+	super.interrupt();
+
+	try {
+	    closeServerSocket();
+	} catch( IOException e ) {
+	    // Ignore
+	}
+
+    }
+
+    private void closeServerSocket() 
+	throws IOException {
+	if( this.serverSocket.isClosed() )
+	    return; // false;
+
+	//try {
+
+	    this.serverSocket.close();
+	    //this.observer.serverSocketClosed( this );
+	    //return true;
+	    
+	    //} catch( IOException e ) {
+	    
+	    //this.observer.serverSocketException( this, e );
+	    //return false;
+
+	    //}
+    }
+
+    public void finalize() {
+	// ..!
+    }
+
+
+
+    public String getStatusString() {
+	StringBuffer b = new StringBuffer();
+	b.append( "bindAddress=" ).append(this.bindAddress.toString() ).append( ",\n" ).
+	    append( "bindPort=" ).append( this.bindPort ).append( ",\n" );
+	this.bindSettingsToSecureStatusString( null, this.bindSettings, b ).append( ",\n" );
+	//b.append( "bindSettings=" ).append( this.bindSettings.toString() ).append( ",\n" ).
+	b.append( "uuid=" ).append( this.uuid );
+	return b.toString();
+    }
+
+    private StringBuffer bindSettingsToSecureStatusString( String childName,
+							   Environment<String,BasicType> child,
+							   StringBuffer b ) {
+
+	// Warning: do NOT print the SSL keyStore- and trustStore passwords!
+	//          (otherwise they would appear in the log files in plain text)
+	if( childName != null 
+	    &&
+	    childName.equalsIgnoreCase(Constants.CONFIG_SERVER_LISTEN_PROPERTY) ) {
+
+	    BasicType wrp_propertyName = child.get( Constants.CONFIG_SERVER_LISTEN_PROPERTY_NAME );
+	    String propertyName        = wrp_propertyName.getString();
+	    if( propertyName.equalsIgnoreCase(Constants.CONFIG_SERVER_SSL_KEYSTOREPASSWORD) 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears your code contains a hardcoded password. Hardcoded secrets or credentials can allow attackers to bypass authentication methods and perform malicious actions. We recommend revoking access to resources using this credential and storing future credentials in a management service such as [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/).

[Learn more about the use of hardcoded credentials](https://cwe.mitre.org/data/definitions/798.html)